daprdocs/content/en/concepts/components-concept.md

```diff
@@ -76,7 +78,19 @@ Lock components are used as a distributed lock to provide mutually exclusive acc
 
 ### Middleware
 
-Dapr allows custom [middleware]({{<ref "middleware.md">}})  to be plugged into the HTTP request processing pipeline. Middleware can perform additional actions on an HTTP request, such as authentication, encryption and message transformation before the request is routed to the user code, or before the response is returned to the client. The middleware components are used with the [service invocation]({{<ref "service-invocation-overview.md">}}) building block.
+Dapr allows custom [middleware]({{<ref "middleware.md">}}) to be plugged into the HTTP request processing pipeline. Middleware can perform additional actions on an HTTP request, such as authentication, encryption and message transformation before the request is routed to the user code, or before the response is returned to the client. The middleware components are used with the [service invocation]({{<ref "service-invocation-overview.md">}}) building block.
 
 - [List of supported middleware components]({{< ref supported-middleware >}})
-- [Middleware implementations](https://github.com/dapr/components-contrib/tree/master/middleware)
\ No newline at end of file
+- [Middleware implementations](https://github.com/dapr/components-contrib/tree/master/middleware)
+
+## Pluggable Components
+
+Dapr allows for users to create their own self-hosted components, they are called pluggable components. These components are just ordinary components that do not need to be written in Go, exist outside the Dapr runtime and are still able to "plug" into Dapr to utilize existing building block APIs.
+
+Creating your own pluggable component can be helpful for scenarios where the Dapr project doesn't provide a component integration you need. Pluggable components are also useful for scenarios where writing a traditional Dapr component might not be feasible (e.g., unfamiliar with Go, remaining untethered to Dapr release cycle, not interested in contributing custom component to Dapr project etc.). For these situations creating a pluggable component could be a better option.
+
+The following pages provide more context on this subject:
+
+- [Pluggable components overview]({{<ref "pluggable-components-overview">}})
+
+**Note:** Since pluggable components are not required to be written in Go, they follow a different implementation process than built-in Dapr components. For more information on that process read the documentation for [Developing new components](https://github.com/dapr/components-contrib/blob/master/docs/developing-component.md)
```

Comment:
```suggestion

{{% alert title="Note" color="primary" %}} Since pluggable components are not required to be written in Go, they follow a different implementation process than built-in Dapr components. For more information on developing built-in components read [developing new components](https://github.com/dapr/components-contrib/blob/master/docs/developing-component.md)
{{% /alert %}}


```